### ISSUE TITLE
"Aux.DateAndTime.SaveAndStampTime" changes the formatting

### ISSUE THREAD

**Issue by castagnotto** [mp4096/aux-funs#1]
The number of blanks between "Last Changed:" and the timestamp seems to be fixed and not depend on the original formatting.

This could probably be changed, right? ;-)
Otherwise it messes up my formatting

**Comment by mp4096** [comment]
Yes, the function writes exactly one whitespace between the matching string and the timestamp.

---

Quick-and-dirty :imp: local workaround: Cf. lines 97--100:

``` MATLAB
if ~isempty(exprMatch)
    % If a match was found, replace this line with a new one
    lines{idx} = [exprMatch{1}, ' ', timestamp];
end
```

You can just change the number of whitespaces in line 99.

---

I'll look into the issue. The problem is, How many whitespaces do you want to have if the line had no timestamp before?

**Comment by castagnotto** [comment]
If it was not defined before

`if isempty(exprMatch)`
I guess the „best“ way would be to check the lines above and/or below for formatting (`nOfCharacters + nOfProminentBlanks = positionAtWhichInformationBegins`) and just choose some spacing if no pattern is recognized (e.g. at the next tab spacing).

Maybe you can use `strrep` if a match is found to change only the timestamp and keep the rest? This would allow information after the timestamp – if wanted – to be preserved.

**Comment by mp4096** [comment]
Hm, I think it is a good idea to replace the timestamp only. `regexprep` would be ideal in this case, for you can then specify the timestamp pattern to be looked for.

I would say, the best way would be the following:
- If the timestamp format is specified in the input arguments, look for it and replace only it
- If no timestamp format was specified:
  - If there is _something_ after the `% Last changed:` tag, then preserve the number of whitespaces after the tag
  - Otherwise, use the default value of one whitespace

Intelligent whitespace pattern search is an overkill in this case, IMO.

**Comment by castagnotto** [comment]
I agree ;-)

what about adding a "\t" instead of a blank? so basically using "sprintf" or similar. Would that be easy to integrate? (cuz in my case it would have solved the issue)

**Comment by mp4096** [comment]
Tabbing is not a problem. One can look for a tag with tabs after it (input argument `format`) and just preserve the tabs or whitespaces, regexps are totally flexible.

MATLAB is totally inconsistent about tabs and whitespaces. In the code, `Ctrl+I` generates no tabs, only whitespaces. In the comments, however, tabs are allowed.
